### PR TITLE
Add prelude items to correct worksheet

### LINKED ISSUE #621: mimic/macro copying preludes to the wrong worksheet
via @rajpurkar

> When I use the macro, instead of adding the heading to the worksheet I'm on, the heading is added to the worksheet the macro is on.

### PR DESCRIPTION
**Pull request by kashizui**
Fixes #621

### CODE REVIEW

File: codalab/lib/bundle_cli.py [hunk 1]
```diff
@@ -2015,7 +2017,8 @@ def newline():
                                 new_bundle_uuids_added.add(new_bundle_uuid)
                                 just_added = True
 
-                    if (item['type'] == worksheet_util.TYPE_MARKUP and item['value'] != '') or item['type'] == worksheet_util.TYPE_DIRECTIVE:
+                    if ((item['type'] == worksheet_util.TYPE_MARKUP and item['value'] != '') or
+                            item['type'] == worksheet_util.TYPE_DIRECTIVE):
```
Comment (kashizui) [inline codalab/lib/bundle_cli.py:2021]: just reindenting here

File: codalab/lib/bundle_cli.py [hunk 2]
```diff
@@ -1697,7 +1697,7 @@ def follow_targets(self, client, bundle_uuid, subpaths):
         while True:
             if not run_finished:
                 info = client.fetch('bundles', bundle_uuid)
-                run_finished = info['state'] in (State.READY, State.FAILED)
+                run_finished = info['state'] in State.FINAL_STATES
```
Comment (kashizui) [inline codalab/lib/bundle_cli.py:1700]: unrelated minor cleanup

File: codalab/lib/bundle_cli.py [hunk 3]
```diff
@@ -2004,6 +2003,9 @@ def newline():
 
                                 # Add prelude
                                 for item2 in prelude_items:
+                                    # Create a copy of the item on the destination worksheet
+                                    item2 = item2.copy()
+                                    item2['worksheet'] = JsonApiRelationship('worksheets', worksheet_uuid)
```
Comment (kashizui) [inline codalab/lib/bundle_cli.py:2008]: this was what was missing and causing the bug

File: codalab/lib/bundle_cli.py [hunk 4]
```diff
@@ -2004,6 +2003,9 @@ def newline():
 
                                 # Add prelude
                                 for item2 in prelude_items:
+                                    # Create a copy of the item on the destination worksheet
+                                    item2 = item2.copy()
```
Comment (percyliang) [inline codalab/lib/bundle_cli.py:2007]: Ah!

**Review by percyliang — APPROVED** [review]
LGTM
